### DISCUSSION TITLE
Azanyum Sazanyum sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/BilsemPython#45] [Soru Cevap] [answered]
Evrenin gizemli bir yerinde Diven diye bir gezegende Azanyum ve Sazanyum adlı iki türlü yaratık yaşıyordu. Her yılın son günü her Azanyum, bir sazanyum ve bir azanyum yaratacak şekilde bölünüyordu. Bir sazanyum ise bir azanyum iki sazanyuma bölünmekteydi. Bu gezegende hayatın tek bir azanyum ile başladığı bilinmekteydi. İncemeler şunu gösterdiki Sazanyum/Azanyum oranı 5/3 ü aştığında bu gezegende yaşam son bulacak. Sizce Diven gezegeninde yaşam bir gün son bulur mu? (Bu arada bölünen Azanyum ve Sazanyumlar yaşamaya devam ediyor)

**Comment by CinarSatir** [comment, marked as answer]
İlk yıllarda oran yükselir ama bir yerden sonra oran 0,00001 ve bu sayıdan kucuk sayılar ile artmaya devam eder bir süre sonra bu oran hiç artmaz bunu ispatlamakiçin yaptıgım program : 
```python
a = 1
s = 0
for i in range(1,11):
    az = a+s
    sa = a+s*2
    a = az+a
    s = sa+s
    print(i,". senede azanyum sayısı",a,"sazanyum sayısı",s,"olur.")
if s/a >=5/3:
    print("Diven gezegeni yok olur")
else:
    print("Diven gezegeni yok olmaz")
```

**Comment by OmerB2** [comment]
İlk yıllarda sayı büyük oranlarda büyür ama sonraki yıllarda artış oranı düşmeye başlar ve çok küçük sayılara düşer 0,0000000001 gibi bir süre sonra sayılar o kadar küçülür ki bu oran hiç artmamaya başlar.
```python
a=1
s=0
for y in range(1,11):
    az=a+s
    sa=a+s*2
    a=a+az
    s=s+sa
    print(y,"Yılında",a,"azanyum",s,"sazanyum olur")
```

**Comment by Sadik1Dev** [comment]
```
azan = 1

sazan = 0

for s in range(1,20):

    azanyum=azan+sazan

    sazanyum=azan+sazan*2

    azan = azan + azanyum

    sazan = sazan + sazanyum

    print(s,"yılında",azan,"kadar azanyum",sazan,"kadar sazanyum vardır")

    if sazan/azan == 5/3:
        print("divan yokoldu")
        break
    elif sazan/azan != 5/3:
        print("divan yok olmaz")
#yok olmazmış
```

**Comment by OmerMete** [comment]
Hocam ben ilk azanyum ve sazanyum sayısı için değişken belirledim. Ondan sonra for un içinde sayılarını bulurken azanyum ve sazanyum sayısı kadar azanyum,azanyum ve sazanyum sayısının iki katı kadar sazanyum oluşur. Forun içinde bunu yaptım ve yazdırdım yıl yıl. Sonda ise if ile gezegenin yok olup olmayacağına baktım ve yok olmuyor.
```
a=1
s=0
for x in range (1,11):
  az=a+s
  saz=a+s*2
  a=az+a
  s=saz+s
  print(x,"senede",a,"kadar azanyum",s,"kadar sazanyum oluşur")
if s/a >=5/3:
  print("diven gezegeni yok oldu")
else:
  print("diven gezegeni yok olomadı")
```
